torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -108,16 +112,17 @@ def _lazy_init(
     # non-root instances
     assert state is root_module
     inconsistent_limit_all_gathers = False
+    inter_node_state = _validate_and_get_hybrid_shard_state(root_module)
     for fsdp_module in state.fsdp_modules(root_module):
         if fsdp_module is root_module:
             continue
 
         if fsdp_module.sharding_strategy in HYBRID_SHARDING_STRATEGIES:
-            _validate_hybrid_shard_setup(state, fsdp_module)
             # Share the allreduce state across FSDP units. This is not strictly necessary
             # as each one already uses the same process group, but can slightly save memory
             # since other FSDP units allreduce state can be garbage collected.
-            fsdp_module._inter_node_state = state._inter_node_state
+            assert inter_node_state is not None, "Implementation error"
```

Comment:
nit: can we have something more descriptive than implementation error, possibly hinting towards what the underlying issue may be?

Reply:
Added a detailed explanation.